Source/Private/ASLayoutTransition.h

```diff
@@ -52,12 +50,12 @@ AS_SUBCLASSING_RESTRICTED
 /**
  * Previous layout to transition from
  */
-@property (nonatomic, readonly) std::shared_ptr<ASDisplayNodeLayout> previousLayout;
+@property (nonatomic, readonly) const ASDisplayNodeLayout &previousLayout;
```

Comment:
Are we sure we are holding on that reference as the transition lifetime?

Reply:
It's an ivar on the transition object (it's copied in `-init`) so it'll live as long as the transition lives.

Reply:
There _is_ a small risk here, which is you read this ref and store it/don't copy it, and then you release the transition while you're holding on to the ref. Our current usage is OK. We could actually change this to be a pointer, and use NS_RETURNS_INNER_POINTER to prevent that … I might try it.

Reply:
Nice